src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.FindSymbols.Finders;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text.Adornments;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    // The VS LSP client supports streaming using IProgress<T> on various requests.
+    // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
+    [ExportLspMethod(LSP.Methods.TextDocumentReferencesName), Shared]
+    internal class FindAllReferencesHandler : IRequestHandler<LSP.ReferenceParams, object[]>
+    {
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FindAllReferencesHandler(IThreadingContext threadingContext)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<object[]> HandleRequestAsync(Solution solution, ReferenceParams referenceParams, ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            // The VS LSP client supports streaming using IProgress<T> on various requests.
+            // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
+            var locations = ArrayBuilder<LSP.Location>.GetInstance();
```

Comment:
```suggestion
            using var _ = ArrayBuilder<LSP.Location>.GetInstance(out var locations);
```

Tihs way we don't leak if canceled.  Note: call .ToArray below, not ToArrayAndFree